tests/Extensions/DontUseThisTestRequest.php

```diff
@@ -63,4 +63,11 @@ protected function attributes(): array
             'age' => 'oldness',
         ];
     }
+
+    protected function withValidator(): void
+    {
+        $this->validator->sometimes('name', 'starts_with:old', function ($input) {
+            return $input->age >= 100;
+        });
+    }
```

Comment:
Don't forget to inform the user about this hidden implementation by describing it to the `readme.md` file.

Also mention that, in this method, be can do both the complex conditional validation as well as the after validation hook.